source/extensions/filters/http/aws_lambda/aws_lambda_filter_config_factory.cc

```diff
@@ -26,11 +26,12 @@ AWSLambdaFilterConfigFactory::createFilterFactoryFromProtoTyped(
                                             context.clusterManager()),
       context.dispatcher(), context.api(), context.threadLocal(), stats_prefix,
       context.scope(), proto_config);
-
+  auto should_propagate_origin = proto_config.propagate_original_routing().value();
   return
-      [&context, config](Http::FilterChainFactoryCallbacks &callbacks) -> void {
+      [&context, should_propagate_origin, config](Http::FilterChainFactoryCallbacks &callbacks) -> void {
```

Comment:
this looks safe to me. just wanted to call out that this is only safe since we captured by value rather than by reference